specification/network/resource-manager/Microsoft.Network/stable/2018-06-01/networkWatcher.json

```diff
@@ -3148,6 +3273,221 @@
         }
       },
       "description": "Connection state snapshot."
+    },
+    "NetworkConfigurationDiagnosticParameters": {
+      "description": "Parameters to get network configuration diagnostic.",
+      "required": [
+        "targetResourceId",
+        "queries"
+      ],
+      "properties": {
+        "targetResourceId": {
```

Comment:
what permissions are needed on the target resource to perform the network configuration diagnostic

Reply:
Probably, read access on NSG.